torch_xla/csrc/tensor_util.cpp

```diff
@@ -106,13 +106,18 @@ bool Use32BitLong() {
   return use_32bit_long;
 }
 
+bool IsTpuDevice(XlaDeviceType hw_type) {
+  // TODO(JackCaoG): We can't assume all SPMD device are TPU deivce.
+  return hw_type == XlaDeviceType::TPU && hw_type == XlaDeviceType::SPMD;
```

Comment:
Wouldn't this always return false?

Reply:
good catch, it should a `||`

Reply:
OK we actually also run SPMD on CPU.. I need to use a better way to tell if underlying type is TPU